### PR TITLE
Add proxy to instructions when creating a token for a node

### LINKED ISSUE #5429: Join links printed by tctl aren't correct for joining via proxy
### Description

**What happened**: Running `tctl tokens add` or `tctl nodes add` on Cloud instances prints an auth server address that's wrong for two reasons:

1) It's an internal pod IP rather than an externally-accessible address (`proxy_service.public_addr`)
2) It uses port 3025 rather than 3080 (it should use the same port as `proxy_service.public_addr`)

**What you expected to happen**: `tctl` and/or the server should know that for Cloud instances, nodes and other services must join via the proxy. It may be that we should add a setting to Teleport which dictates whether nodes are expected to join via auth or proxy, then set this printed address accordingly.

See diff below for changes that should happen.

```diff
$ tctl tokens add --type=node --ttl=1h

The invite token: 238179057b40cde5eca6647352a24fae
This token will expire in 60 minutes

Run this on the new node to join the cluster:

 teleport start \
   --roles=node \
   --token=238179057b40cde5eca6647352a24fae \
   --ca-pin=sha256:deaaec29ace4e8584606015dfa682501703141ea979fa5d30a1d5085097e2c9e \
-   --auth-server=100.96.82.200:3025
+    --auth-server=example.teleport.sh:443

Please note:

  - This invitation token will expire in 60 minutes
-  - 100.96.82.200:3025 must be reachable from the new node
+ - example.teleport.sh:443 must be reachable from the new node
```

**How to reproduce it (as minimally and precisely as possible)**: Run `tctl tokens add --type=node` on a Cloud instance.

### Environment
- Teleport version (use `teleport version`): Teleport 5.1.0

### LINKED ISSUE #6320: tctl nodes add output issues
### Description

**What happened**:

The output of `tctl nodes add` gives instructions to run on a node that will not result in the expected connection for the node.

ex output of `tctl nodes add`:
```bash
Run this on the new node to join the cluster:

> teleport start \
   --roles=app \
   --token=tokenvalue \
   --ca-pin=sha256:pinnumber \
   --auth-server=172.1.6.1:3025
```

Specifically:
  - gives the auth server as a machine ip that is often not directly available to a node.  Would expect the public address specified or a method of using the proxy address: port.
  - in a cloud and the roles `db`, `kube` and `app` context does not give the proxy address and port that is required
  - For roles such as `db`, `kube`, and `app` the command would likely do nothing unless a `teleport.yaml` is already configured.  No info given on that.

If a admin wants to add a node with a particular role they run `tctl nodes add 

**What you expected to happen**:

More context is given to the user or the system can provide commands that will work in the Teleport instance environment.

### Reproduction Steps
As minimally and precisely as possible, describe step-by-step how to reproduce the problem.
1.Login via `tsh` to a cloud environment
2.Run `tctl nodes add`
3.`teleport` start command will not work in a cloud env

### Server Details
- Teleport version (run `teleport version`): 6.0.2
- Server OS (e.g. from `/etc/os-release`): ubuntu
- Where are you running Teleport? (e.g. AWS, GCP, Dedicated Hardware): AWS

### PR DESCRIPTION
**Pull request by mcbattirola**
This PR changes the instructions printed to the user when running `tokens add type=node` to print the proxy as the `--auth-server` .

Currently `tokens add type=node` is outputting the auth server to the user instead of the proxy server:

```bash
$ tctl tokens add --type=node --identity=./id --auth-server=matheus13.cloud.gravitational.io:443

Run this on the new node to join the cluster:

> teleport start \
   --auth-server=100.92.7.4:3025
...
```

This PR changes it to print the proxy as the `auth-server`:

```bash
> teleport start \
   --auth-server=matheus13.cloud.gravitational.io:443
```

Adding a token with `type=app` and `type=db` already prints the proxy server address. I used the same logic here.

I left the auth server as a fallback, but I'm not sure if it is possible to not have a proxy since for DB and apps we error out when there are no proxies. Should we do the same here?

closes #5429 
closes #6320

### CODE REVIEW

File: tool/tctl/common/token_command.go [hunk 1]
```diff
@@ -218,12 +218,23 @@ func (c *TokenCommand) Add(client auth.ClientI) error {
...
+		if err != nil {
+			return trace.Wrap(err)
+		}
+
+		authServer := authServers[0].GetAddr()
+
+		if len(proxies) != 0 {
+			authServer = proxies[0].GetPublicAddr()
+		}
+
```
Comment (russjones) [inline tool/tctl/common/token_command.go:231]: Does this mean we'll always switch to displaying the proxy as how to join the cluster? Not sure we want to do that, joining the cluster via the auth server is a legitimate use case.

Comment (mcbattirola) [reply tool/tctl/common/token_command.go:231]: I see, we don't wanna do that then.

Do you think we should add a new config variable and have the server signal to the client that it should join via proxy?
Another way I can think of is printing the URL the client actually used to authenticate.

To give you context, the problem is that the auth server is not always reachable to the node, especially on the cloud.

Please let me know your thoughts

Comment (pierrebeaucamp) [reply tool/tctl/common/token_command.go:231]: Sorry for commenting this late on the PR.

I think we might be able to work around this by looking at the Teleport license and default to returning the proxy address if we're running in Cloud.

**Comment by mcbattirola** [comment]
Okay, I reopened the PR with @pierrebeaucamp's suggestion.

I tried to make the smallest change possible:
`tctl tokens add --type=node` will ping the auth server and get its feature flag. If it is a cloud instance, we print the proxy server.
This means we make another request, which adds latency.

I think we could prevent this by changing this [request to check connectivity](https://github.com/gravitational/teleport/blob/master/tool/tctl/common/tctl.go#L209), which currently ignores the response, to call `Ping` and store the result (but that would be a bigger change). 

Please let me know your thoughts

**Comment by russjones** [comment]
Seems reasonable to me. @r0mant @pierrebeaucamp can you two review as well?

**Review by pierrebeaucamp — APPROVED** [review]

**Review by zmb3 — COMMENTED** [review]
Looks fine. Does this logic kick in for `tctl nodes add` too, or do we need to make a separate update for that command?

**Comment by mcbattirola** [comment]
> Looks fine. Does this logic kick in for `tctl nodes add` too, or do we need to make a separate update for that command?

No, `tctl nodes add` is a different flow

I'll update it too, thanks

**Review by zmb3 — APPROVED** [review]

**Comment by webvictim** [comment]
@russjones @zmb3 It would honestly be awesome if we set this flag by default for new Teleport installations. People installing Teleport inside Kubernetes or Terraform often hit this same issue, where the auth server's IP/port are not exposed outside but the proxy's host/port always is.

I'd love to see this set as default behaviour and the some kind of configuration parameter exposed in the Teleport config to use that legacy join behaviour for bigger customers.

**Pull request merged**